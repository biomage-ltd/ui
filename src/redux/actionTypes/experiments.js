const EXPERIMENTS = 'experiments';

/**
 * Create experiments
 */
const EXPERIMENTS_CREATE = `${EXPERIMENTS}/create`;

/**
 * Update experiments
 */
const EXPERIMENTS_UPDATE = `${EXPERIMENTS}/update`;

/**
 * Saving experiment
 */
const EXPERIMENTS_SAVING = `${EXPERIMENTS}/saving`;

/**
 * Experiment saved
 */
const EXPERIMENTS_SAVED = `${EXPERIMENTS}/saved`;

/**
 * Update experiments
 */
const EXPERIMENTS_ERROR = `${EXPERIMENTS}/error`;

export {
  EXPERIMENTS_CREATE,
  EXPERIMENTS_UPDATE,
  EXPERIMENTS_ERROR,
  EXPERIMENTS_SAVING,
  EXPERIMENTS_SAVED,
};
