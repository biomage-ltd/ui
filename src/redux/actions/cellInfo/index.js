import setCellInfoFocus from './setCellInfoFocus';
import updateCellInfo from './updateCellInfo';

export {
  setCellInfoFocus,
  updateCellInfo,
};
