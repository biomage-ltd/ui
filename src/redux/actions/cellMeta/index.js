import loadCellMeta from './loadCellMeta';

export default loadCellMeta;
