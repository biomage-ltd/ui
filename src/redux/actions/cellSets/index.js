import createCellSet from './createCellSet';
import deleteCellSet from './deleteCellSet';
import resetCellSets from './resetCellSets';

import loadCellSets from './loadCellSets';
import saveCellSets from './saveCellSets';
import updateCellSetHierarchy from './updateCellSetHierarchy';
import updateCellSetProperty from './updateCellSetProperty';
import updateCellSetSelected from './updateCellSetSelected';
import setCellSetHiddenStatus from './setCellSetHiddenStatus';
import unhideAllCellSets from './unhideAllCellSets';

export {
  createCellSet,
  deleteCellSet,
  resetCellSets,
  loadCellSets,
  saveCellSets,
  updateCellSetHierarchy,
  updateCellSetSelected,
  updateCellSetProperty,
  setCellSetHiddenStatus,
  unhideAllCellSets,
};
