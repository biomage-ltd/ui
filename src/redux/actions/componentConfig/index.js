import loadPlotConfig from './loadPlotConfig';
import loadComponentConfig from './loadComponentConfig';
import updatePlotConfig from './updatePlotConfig';
import savePlotConfig from './savePlotConfig';

export {
  loadPlotConfig,
  loadComponentConfig,
  updatePlotConfig,
  savePlotConfig,
};
