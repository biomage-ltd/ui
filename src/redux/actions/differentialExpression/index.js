import loadDifferentialGeneExpression from './loadDifferentialExpression';
import setComparisonType from './setComparisonType';
import setComparisonGroup from './setComparisonGroup';

export {
  loadDifferentialGeneExpression,
  setComparisonType,
  setComparisonGroup,
};
