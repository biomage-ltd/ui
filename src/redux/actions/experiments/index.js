import createExperiment from './createExperiment';
import saveExperiment from './saveExperiment';
import updateExperiment from './updateExperiment';
import loadExperiments from './loadExperiments';
import deleteExperiments from './deleteExperiments';

export {
  createExperiment,
  saveExperiment,
  updateExperiment,
  loadExperiments,
  deleteExperiments,
};
