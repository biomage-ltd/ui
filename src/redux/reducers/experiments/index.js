import initialState from './initialState';
import {
  EXPERIMENTS_CREATED,
  EXPERIMENTS_UPDATED,
  EXPERIMENTS_LOADED,
  EXPERIMENTS_LOADING,
  EXPERIMENTS_ERROR,
  EXPERIMENTS_SAVED,
  EXPERIMENTS_SAVING,
} from '../../actionTypes/experiments';
import experimentsCreate from './experimentsCreate';
import experimentsUpdate from './experimentsUpdate';
import experimentsLoading from './experimentsLoading';
import experimentsLoaded from './experimentsLoaded';
import experimentsError from './experimentsError';
import experimentsSaving from './experimentsSaving';
import exprimentsSaved from './experimentsSaved';

const experimentsReducer = (state = initialState, action) => {
  switch (action.type) {
    case EXPERIMENTS_CREATED: {
      return experimentsCreate(state, action);
    }

    case EXPERIMENTS_UPDATED: {
      return experimentsUpdate(state, action);
    }

    case EXPERIMENTS_LOADING: {
      return experimentsLoading(state, action);
    }

    case EXPERIMENTS_LOADED: {
      return experimentsLoaded(state, action);
    }

    case EXPERIMENTS_ERROR: {
      return experimentsError(state, action);
    }

    case EXPERIMENTS_SAVING: {
      return experimentsSaving(state, action);
    }

    case EXPERIMENTS_SAVED: {
      return exprimentsSaved(state, action);
    }
    default: {
      return state;
    }
  }
};

export default experimentsReducer;
