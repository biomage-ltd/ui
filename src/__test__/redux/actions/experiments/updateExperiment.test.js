import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import {
  EXPERIMENTS_UPDATED,
} from '../../../../redux/actionTypes/experiments';
import { updateExperiment, saveExperiment } from '../../../../redux/actions/experiments';
import initialExperimentState, { experimentTemplate } from '../../../../redux/reducers/experiments/initialState';

jest.mock('localforage');

jest.mock('../../../../redux/actions/experiments/saveExperiment');
saveExperiment.mockImplementation(() => async () => { });

const mockStore = configureStore([thunk]);

describe('updateExperiment', () => {
  const experimentId = 'experiment-1';

  const mockExperiment = {
    ...experimentTemplate,
    name: 'experiment-1',
    id: experimentId,
  };

  const updatedExperiment = {
    ...mockExperiment,
    name: 'updated-experiment',
  };

  const mockState = {
    experiments: {
      ...initialExperimentState,
      [experimentId]: mockExperiment,
    },
  };

  it('Dispatches action when called', async () => {
    const store = mockStore(mockState);
    await store.dispatch(updateExperiment(experimentId, updatedExperiment));

    const actions = store.getActions();

    expect(actions[0].type).toEqual(EXPERIMENTS_UPDATED);

    expect(actions).toMatchSnapshot();
  });

  it('Dispatches call to save experiment', async () => {
    const store = mockStore(mockState);
    await store.dispatch(updateExperiment(experimentId, updatedExperiment));

    expect(saveExperiment).toHaveBeenCalled();
  });
});
