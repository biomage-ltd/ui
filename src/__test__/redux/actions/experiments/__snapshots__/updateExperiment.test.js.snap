// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`updateExperiment Dispatches action when called 1`] = `
Array [
  Object {
    "payload": Object {
      "experiment": Object {
        "createdAt": null,
        "description": "Analysis description",
        "id": "experiment-1",
        "lastViewed": null,
        "name": "updated-experiment",
        "projectUuid": null,
      },
      "experimentId": "experiment-1",
    },
    "type": "experiments/updated",
  },
]
`;
